l1-contracts/contracts/bridgehub/STMDeploymentTracker.sol

```diff
@@ -32,6 +32,13 @@ contract STMDeploymentTracker is ISTMDeploymentTracker, ReentrancyGuard, Ownable
         _;
     }
 
+    /// @notice Checks that the message sender is the bridgehub.
+    modifier onlyOwnerViaRouter(address _prevMsgSender) {
+        // solhint-disable-next-line gas-custom-errors
+        require(msg.sender == address(SHARED_BRIDGE) && _prevMsgSender == owner(), "STM DT: not owner via router");
```

Comment:
btw, maybe rename the `SHARED_BRIDGE` to `ASSET_ROUTER`? Or even `L1_ASSET_ROUTER`?